src/javascript/app_2/Stores/Modules/Trading/trade_store.js

```diff
@@ -448,6 +448,12 @@ export default class TradeStore extends BaseStore {
 
     @action.bound
     changeAllowEquals() {
+        if (!this.contract_types_list
+            || !this.contract_type
+            || !this.duration_unit
+            || !this.contract_start_type
+            || !this.expiry_type) return;
```

Comment:
I think it's cleaner if you add these checks to the functions where these variables are used?